specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-05-13-preview/desktopvirtualization.json

```diff
@@ -3637,181 +3966,496 @@
+                "value": "Completed",
+                "description": "Hostpool update completed."
               }
             ]
           },
           "type": "string",
-          "description": "The type of image session hosts use in the hostpool.",
+          "description": "State of hostpool update.",
           "x-nullable": false
         },
-        "marketPlaceInfo": {
-          "$ref": "#/definitions/MarketPlaceInfoProperties",
-          "description": "The values to uniquely identify a gallery image."
-        },
-        "storageBlobUri": {
+        "timeCreated": {
+          "format": "date-time",
           "type": "string",
-          "description": "The uri to the storage blob which contains the VHD. Image type must be StorageBlob."
+          "description": "The hostpool update create time."
         },
-        "customId": {
-          "type": "string",
-          "description": "The resource id of the custom image or shared image. Image type must be custom."
-        }
-      }
-    },
-    "MarketPlaceInfoProperties": {
-      "description": "Image configurations of HostPool.",
-      "type": "object",
-      "properties": {
-        "offer": {
+        "timeStarted": {
+          "format": "date-time",
           "type": "string",
-          "description": "The offer of the image."
+          "description": "The hostpool update start time."
         },
-        "publisher": {
+        "timeFailed": {
+          "format": "date-time",
           "type": "string",
-          "description": "The publisher of the image."
+          "description": "The hostpool update fail time."
         },
-        "sku": {
+        "timeEnded": {
+          "format": "date-time",
           "type": "string",
-          "description": "The sku of the image."
+          "description": "The hostpool update end time."
         },
-        "exactVersion": {
-          "type": "string",
-          "description": "The exact version of the image."
+        "listOfErrors": {
+          "type": "array",
+          "description": "The alerts given to customers for hostpool update.",
+          "items": {
+            "$ref": "#/definitions/HostPoolUpdateFault"
+          }
+        },
+        "progressPercentage": {
+          "type": "number",
+          "format": "float",
+          "description": "The progress percentage."
+        },
+        "sessionHostsToMigrate": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of session hosts to migrate by hostpool update."
+        },
+        "sessionHostsMigrating": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of session hosts migrating by hostpool update."
+        },
+        "sessionHostsMigrated": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The number of session hosts migrated by hostpool update."
+        },
+        "sessionHostsRollbackFailed": {
```

Comment:
Just use "sessionHostsFailed"; "sessionHostsRollbackFailed" is confusing, does it mean rollback failed or update failed?

Reply:
@shailahirji they came up with the names

Reply:
rollback failed

Reply:
Should we rename it to "sessionHostsFailed"?

Reply:
so keep? because it gives more infor this is for when a rollback has failed

Reply:
To clarify, we have numbers for "sessionHostsToUpdate", "sessionHostsUpdating", "sessionHostsUpdated", and "sessionHostsRollbackFailed".  But sum of "sessionHostsUpdating", "sessionHostsUpdated", and "sessionHostsRollbackFailed" is not equal to "sessionHostsToUpdate" because we don't have number for "sessionHostsFailed"/"sessionHostsUpdateFailed". add this number? @shailahirji what do you think?